features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/data/AuthManager.jsx

```diff
@@ -177,8 +177,13 @@ class AuthManager {
         if (user) {
             const userScopes = user.scopes;
             const validScope = APIClient.getScopeForResource(resourcePath, resourceMethod);
-            return validScope.then((scope) => {
-                return userScopes.includes(scope);
+            let validity = []
+            return validScope.then((scopes) => {
+                scopes.map(scope => {
+                    const isValid = userScopes.includes(scope);
```

Comment:
This iterates through the entire scopes set even though we need to include only one of the defined scopes